### DISCUSSION TITLE
Feature comparison between the existing features of YASA and the current official Admin UI

### DISCUSSION THREAD

**Discussion by almogtavor** [yasa-org/yasa#59] [General]
It's pretty hard to keep on track for what is needed to do next. I think such a comparison can clarify what the missions are and help us start working on filling the gap of features between YASA and the current admin UI.

Is there any comparison between the existing features of YASA vs the current official Admin UI?

**Comment by kezhenxu94** [comment]
@epugh had drafted a doc to compare the features. https://docs.google.com/spreadsheets/d/13tjLXZezQmnRh2iSquj51a3VAUklUa_e6pVLLVu8sUM/edit?usp=sharing

**Comment by epugh** [comment]
I ran out of steam when doing the feature comparisoin, because I realized I was going pretty low level, and it might never get to done!   I think the Feature Comparison becomes a lot more interesting if we start seeing new features showing up to Yasa.

I think there are two strategies for Yasa.  The first is to achieve parity with existing Solr Admin.    The other way is to add NEW Admin related features to Yasa that don't exist in Solr Admin.  I added a tab with "Missing Admin Features" that I put two ideas in.   I'm thinking that we could look at the APIs that Solr has, and identify which ones you can only access via CURL etc, and maybe those are candidates for Yasa to have features that Solr Admin doesn't have, which would encourage adoption.

**Reply by epugh** [reply]
@almogtavor if you are going to keep pushing on the parity work, I'll happily test it and start updating that spreadsheet more ;-)

**Comment by epugh** [comment]
Also, I'm going to work on integrating Yasa plugin into the Chorus Project in the next few days as well since we seem to be starting to move again.  https://github.com/querqy/chorus

@almogtavor I find this a great way to test out new UI features, to use them in a "real" reference implementation project.

**Comment by epugh** [comment]
@almogtavor I wanted to follow up with you after the conversation on the Solr JIRA...   I'm feeling like the blocker to moving forward is more the core "this needs to work for me to install Yasa" in my everyday projects (like Chorus and my client work!)...    Authentication is I think the biggest one, and I believe that @kezhenxu94 made some progress. 

What  do you think about leading the effort on that, and I'll happily review/commit the fixes o n the Solr side.   I've been building my knowledge of solr internals over the past year, so I think I'm in a better place to support you then I was siz months ago....

**Reply by kezhenxu94** [reply]
Basic authorization is supported already and JWT authorization needs Solr side changes. See https://issues.apache.org/jira/plugins/servlet/mobile#issue/SOLR-15296

**Reply by epugh** [reply]
Well that's embarrasing ;-)   I just assigned the ticket to me to track it better.   

Since this is a bit of a chicken/egg  (to use the colloquism https://www.merriam-webster.com/dictionary/chicken-and-egg), do we have a branch of YASA with whatever  JWT changes that are needed, so I could stand up Solr and YASA and work on the whitelisting?

**Reply by kezhenxu94** [reply]
> Well that's embarrasing ;-)   I just assigned the ticket to me to track it better.   
> 
> 
> 
> Since this is a bit of a chicken/egg  (to use the colloquism https://www.merriam-webster.com/dictionary/chicken-and-egg), do we have a branch of YASA with whatever  JWT changes that are needed, so I could stand up Solr and YASA and work on the whitelisting?

This was also already prepared https://github.com/yasa-org/yasa/tree/auth/test/docker

**Comment by kezhenxu94** [comment]
Do we have any progress here?

**Reply by almogtavor** [reply]
I think we should prioritize some features to start with.

**Reply by epugh** [reply]
I've been mulling over this, and I think the biggest gap we have in existing Solr is that the Query interface is built around the old V1 query structures, and there is nothing that supports the https://solr.apache.org/guide/8_5/json-request-api.html and https://solr.apache.org/guide/8_5/json-facet-api.html, which means most users aren't embracing them.  

We could take some visual inspiration from some of the Elasticsearch/Kibana interfaces and properly support those syntaxes.....
